pkg/config/msp.go

```diff
@@ -186,19 +186,100 @@ func GetMSPConfigurationForConsortiumOrg(config *cb.Config, consortiumName, orgN
 		return MSP{}, fmt.Errorf("consortium org %s does not exist in config", orgName)
 	}
 
-	return getMSPConfigForOrg(consortiumOrgGroup, orgName)
+	return getMSPConfig(consortiumOrgGroup)
 }
 
-// getMSPConfigForOrg parses the MSP value in a config group for a specific org and returns
+// AddRootCAToMSP takes a root CA x509 certificate and adds it to the
+// list of rootCerts for the specified application org MSP.
+func (c *Config) AddRootCAToMSP(rootCA *x509.Certificate, orgName string) error {
+	if (rootCA.KeyUsage & x509.KeyUsageCertSign) == 0 {
+		return errors.New("certificate KeyUsage must be x509.KeyUsageCertSign")
+	}
+
+	if !rootCA.IsCA {
+		return errors.New("certificate must be a CA certificate")
+	}
+
+	org, err := getApplicationOrg(c.Updated, orgName)
+	if err != nil {
+		return err
+	}
+
+	fabricMSPConfig, err := getFabricMSPConfig(org)
+	if err != nil {
+		return fmt.Errorf("getting msp config: %v", err)
+	}
+
+	certBytes := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: rootCA.Raw})
+
+	fabricMSPConfig.RootCerts = append(fabricMSPConfig.RootCerts, certBytes)
+
+	err = addMSPConfigToOrg(org, fabricMSPConfig)
+	if err != nil {
+		return fmt.Errorf("adding msp config to org: %v", err)
+	}
+
+	return nil
+}
+
+// YEAR is a time duration for a standard 365 day year.
+const YEAR = 365 * 24 * time.Hour
+
+// RevokeCertificateFromMSP takes a variadic list of x509 certificates, creates
+// a new CRL signed by the specified ca certificate and private key, and appends
+// it to the revocation list for the specified application org MSP.
+func (c *Config) RevokeCertificateFromMSP(orgName string, caCert *x509.Certificate, caPrivKey *ecdsa.PrivateKey, certs ...*x509.Certificate) error {
```

Comment:
A few things

- I missed the singular/plural of 'RevokeCertificate' - it's variadic so certificate should be plural
- The name implies something more imperative than I believe it does. This really creates and adds a CRL.
- This is another place where a mutable MSP would be conceptually simpler to deal with

Even though this is exactly what the story asked for, stepping back, I wonder if we just want to provide a way to add a CRL and a function that creates a CRL with a SigningIdentity.

Reply:
I'll create a story for this MSP-related work.